### DISCUSSION TITLE
Provider station is hogging trains because it is "full" but they wait to leave because requester is also "full"

### DISCUSSION THREAD

**Discussion by adamk33n3r** [Opinionated-Blueprints/10-Books-Full-of-Rails#92] [Q&A] [answered]
I'm just starting out with trains and this blueprint book. I have a provider station at iron ore that has a 3 stack, so it can support up to 4 trains. I have a requester station for iron ore. I'm not using the iron ore that quickly, so what ends up happening is a train is unloading at the station but is backed up because the buffer chests are full. Then I have a train at the provider station sitting there waiting for the requester station to be empty, along with a few more trains in the stacker behind it.

Did I mess something up? I feel like even if the provider station has enough items to support 4 trains to come pick stuff up, it should only send 1 at most if the destination is full. Is there something I can do? Or do I need to just make WAY more trains and depots?

**Comment by Bocian-1** [comment, marked as answer]
You need a lot of trains, but depots should be a rarity.
If a train goes to a requester and is unable to unload completely at once, then you set something wrong when placing the requester.
https://github.com/Opinionated-Blueprints/10-Books-Full-of-Rails/wiki/Provider-and-Requester-stations

**Reply by adamk33n3r** [reply]
Thank you so much for the reply! Needing more trains was definitely the issue. I was perhaps used to LTN and the way it did things, or maybe it's just been too long haha. But adding enough trains for every stacker fixed the issue. And I think the other issue was just a weird edge case.
